source/PlayerInfo.cpp

```diff
@@ -3523,6 +3573,39 @@ void PlayerInfo::Save(const string &path) const
 		mission.Save(out);
 	for(const Mission &mission : inactiveMissions)
 		mission.Save(out);
+	map<string, map<string, int>> offWorldMissionCargo;
+	map<string, map<string, int>> offWorldMissionPassengers;
+	for(const auto &it : ships)
+	{
+		const Ship &ship = *it.get();
+		// If the ship is at the player's planet, its mission cargo allocation does not need to be saved.
+		if(ship.GetPlanet() == planet)
+			continue;
+		for(const auto &cargo : ship.Cargo().MissionCargo())
+			offWorldMissionCargo[cargo.first->UUID().ToString()][ship.UUID().ToString()] = cargo.second;
+		for(const auto &passengers : ship.Cargo().PassengerList())
+			offWorldMissionPassengers[passengers.first->UUID().ToString()][ship.UUID().ToString()] = passengers.second;
+	}
+	auto SaveMissionCargoDistribution = [&out](map<string, map<string, int>> toSave, bool passengers) -> void
+	{
+		if(passengers)
+			out.Write("mission passengers");
+		else
+			out.Write("mission cargo");
+		out.BeginChild();
+		out.Write("player ships");
+		out.BeginChild();
+		for(const auto &it : toSave)
+			for(const auto &it2 : it.second)
+				out.Write(it.first, it2.first, it2.second);
+		out.EndChild();
```

Comment:
The convention elsewhere is to add an indent after calling BeginChild and remove one before calling EndChild, while wrapping everything in braces. Also, instead of "it" and "it2," I think we typically use "oit" for "other it" or "sit" for "second it."
```suggestion
		{
			out.Write("player ships");
			out.BeginChild();
			{
				for(const auto &it : toSave)
					for(const auto &sit: it.second)
						out.Write(it.first, sit.first, sit.second);
			}
			out.EndChild();
		}
```